src/platform/tls_schannel.c

```diff
@@ -612,6 +615,56 @@ typedef struct CXPLAT_TLS {
     //
     QUIC_TLS_SECRETS* TlsSecrets;
 
+    //
+    // Session state and resumption have traditionally been managed by the TLS layer without
+    // any intervention from the application layer. Managed application session state enhances this functionality
+    // by allowing the application to include payload in the resumption data as well as to control/participate
+    // in the session state management process.
+    //
+    // SChannel TLS implementation in Server2025 and newer releases includes this functionality. The main caveat
+    // is that SChannel requires the application to take ownership of session state management in order to use this feature.
+    // SChannel refers to encrypted client-side resumption token as "Session Ticket" and the server-side clear-text app session state
+    // as "App Session State". MsQuic refers to the latter as "ticket data" in some places.
+    //
+    // MsQuic uses a single callback function to notify the application of incoming App Session State or Session Ticket,
+    // relying on the context to determine what type of data is being handled.
+    //
+    // This TLS state structure represents either the client side or the server side of the TLS connection.
+    // The server side handles the plain-text application session state, which sometimes saved in the server instances of this struct.
+    // The client side handles only the encrypted session ticket, which is sometimes stored on the client instances of this struct.
+    //
+
+    //
+    // Length of allocated application state buffer on the server
+    //
+    uint32_t RxAppSessionStateAllocLength;
+
+    //
+    // Received application session state on the server, typically notified to the server app.
+    //
+    PSEC_APP_SESSION_STATE RxAppSessionState;
```

Comment:
Can we have the "client" and "server" in the name for these variables? It's a little difficult to follow when looking at how they are used.

Maybe we can match the bit flags defined above?